core/src/main/java/org/apache/carbondata/core/datamap/DataMapUtil.java

```diff
@@ -91,43 +94,129 @@ public static DataMapJob getDataMapJob(Configuration configuration) throws IOExc
    * @param carbonTable
    * @throws IOException
    */
-  public static void executeDataMapJobForClearingDataMaps(CarbonTable carbonTable)
-      throws IOException {
-    String dataMapJobClassName = "org.apache.carbondata.spark.rdd.SparkDataMapJob";
+  private static void executeDataMapJobForClearingSegmentDataMaps(DataMapJob dataMapJob,
+      CarbonTable carbonTable) throws IOException {
+    SegmentStatusManager.ValidAndInvalidSegmentsInfo validAndInvalidSegmentsInfo =
+            getValidAndInvalidSegments(carbonTable, FileFactory.getConfiguration());
+    List<String> invalidSegment = new ArrayList<>();
+    for (Segment segment : validAndInvalidSegmentsInfo.getInvalidSegments()) {
+      invalidSegment.add(segment.getSegmentNo());
+    }
+    DistributableDataMapFormat dataMapFormat = new DistributableDataMapFormat(carbonTable, null,
+        validAndInvalidSegmentsInfo.getValidSegments(), invalidSegment, null, true,
+        DataMapLevel.FG);
+    dataMapJob.execute(dataMapFormat);
+  }
+
+  public static void clearEmbeddedDataMaps(CarbonTable carbonTable) throws IOException {
+    String dataMapJobClassName = "org.apache.carbondata.indexserver.EmbeddedDataMapJob";
     DataMapJob dataMapJob = (DataMapJob) createDataMapJob(dataMapJobClassName);
     if (dataMapJob == null) {
       return;
     }
-    String className = "org.apache.carbondata.core.datamap.DistributableDataMapFormat";
-    SegmentStatusManager.ValidAndInvalidSegmentsInfo validAndInvalidSegmentsInfo =
-        getValidAndInvalidSegments(carbonTable, FileFactory.getConfiguration());
-    List<Segment> validSegments = validAndInvalidSegmentsInfo.getValidSegments();
-    List<Segment> invalidSegments = validAndInvalidSegmentsInfo.getInvalidSegments();
-    DataMapExprWrapper dataMapExprWrapper = null;
-    if (DataMapStoreManager.getInstance().getAllDataMap(carbonTable).size() > 0) {
-      DataMapChooser dataMapChooser = new DataMapChooser(carbonTable);
-      dataMapExprWrapper = dataMapChooser.getAllDataMapsForClear(carbonTable);
-    } else {
+    executeDataMapJobForClearingSegmentDataMaps(dataMapJob, carbonTable);
+  }
+
+  public static void clearDistributedDataMaps(CarbonTable carbonTable)
+      throws IOException {
+    String dataMapJobClassName = "org.apache.carbondata.indexserver.DistributedClearCacheJob";
```

Comment:
Move this class name to a constant field in the same class

Reply:
DistributedClearCacheJob is in spark2 module. cannot access the class from here(core)

Reply:
There is a method in `DataMapUtil` class called `clearEmbeddedDataMaps` which is doing the same work as `clearDistributedDataMaps`. It would be better if we change the method signature to add one more argument for className and rename the method `clearEmbeddedDataMaps` to a more generic name